float8_experimental/float8_dynamic_linear.py

```diff
@@ -30,18 +30,17 @@ def forward(
 
     @staticmethod
     def backward(ctx, gradY):
-        gradY_scale = tensor_to_scale(gradY, torch.float8_e5m2)
-        fp8_tensor = to_fp8_no_autograd(
-            gradY, gradY_scale, torch.float8_e5m2, ctx.emulate
-        )
+        fp8_dtype = torch.float8_e5m2fnuz if IS_AMD else torch.float8_e5m2
```

Comment:
should this be configurable in the forward with a reasonable default?

Reply:
I wonder if it would be better to have `torch.backends.[cuda|hip|mps].supports_dtype(XYZ)` API, because I assume XPUs would probably use `fnuz` flavor, but say ARM would be using `e5m2` flavor

Reply:
I think this should be configurable instead of depending on env, numerics should be as predictable as possible. It's also valuable to emulate numerics without having the hardware, for debugging.

Reply:
Oaky, so it sounds like we want this to be defined at module construction.  It is up to the constructor of the module to ensure that _scaled_mm will work with  their module.

I do think that Nikitas backend dtype helper could a useful pytorch feature, not sure if that should live here though